### ISSUE TITLE
`invalid-legacy-type-variable` - false negative on redefined `TypeVar`

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#3732]
### Summary

`main.py`:

```py
from typing import TypeVar

T = TypeVar("T")
T = TypeVar("T")  # no error
```

`ty.json`:

```json
{
    "environment": {
        "python-version": "3.14"
    },
    "rules": {
        "all": "error"
    }
}
```

[Playground](https://play.ty.dev/d4fc6058-c774-4ab8-bd3d-919534ae389f)

[the docs](https://docs.astral.sh/ty/reference/rules/#invalid-legacy-type-variable) say this should be an error

### Version

267af002f

**Comment by dhruvmanila** [comment]
Thanks for the report. This seems like an oversight. It's currently a TODO (https://github.com/astral-sh/ruff/blob/8c800ba3d5f1463639cc875ae1cbc01bc2537579/crates/ty_python_semantic/resources/mdtest/generics/legacy/variables.md?plain=1#L105-L114), we should either update the docs or just implement it.
